addons/l10n_in_ewaybill_stock/models/stock_move.py

```diff
@@ -0,0 +1,56 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+    _description = "Stock Move Ewaybill"
+
+    l10n_in_ewaybill_id = fields.One2many(related="picking_id.l10n_in_ewaybill_id")
+
+    company_currency_id = fields.Many2one(related='company_id.currency_id')
+
+    # Need to store values because we send it to the ewaybill and we need to keep the same value
+    ewaybill_price_unit = fields.Monetary(
+        compute='_compute_ewaybill_price_unit',
+        currency_field='company_currency_id',
+        store=True
+    )
+    ewaybill_tax_ids = fields.Many2many(
+        comodel_name='account.tax',
+        string="Taxes",
+        compute='_compute_tax_ids',
+        store=True
+    )
+
+    @api.depends('product_id.uom_id', 'product_id.standard_price', 'l10n_in_ewaybill_id')
+    def _compute_ewaybill_price_unit(self):
+        for line in self.filtered(lambda line: line.l10n_in_ewaybill_id.state == 'pending'):
+            line.ewaybill_price_unit = line.product_id.uom_id._compute_price(
```

Comment:
@hamo-odoo This logic could be in l10n_in_stock as well (but better in a separate method.  Better to avoid inheriting from the _compute method).  Maybe we can do the api.depends with a lambda.

Reply:
Not sure for the lambda function :\ if all fields would be from product_id I would had agreed to it

Reply:
@api.depends(lambda self: ['product_id.uom_id', 'product_id.standard_price', 'l10n_in_ewaybill_id'] + self._get_other_ewaybill_deps()])  

would not work?

Reply:
It would work but I don't see any benefit while passing a function (I don't see `_get_other_ewaybill_deps`)

If you can explain me why it would be better here to pass a function rather than passing dot-separated sequence of field names? I would like to change it after knowing it :wink:

Reply:
Because it is difficult to inherit this compute directly in the other modules as it would need to be defined in l10n_in_stock where we can not store it.

Reply:
Created `_l10n_in_compute_xx` for logic kept `_compute_l10n_in_xx` for actual compute